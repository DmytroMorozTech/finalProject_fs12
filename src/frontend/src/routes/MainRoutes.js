import React from 'react'
import { Route, Switch } from 'react-router-dom'
import Messages from '../components/Main/Messages/Messages'
import Profile from '../components/Main/Profile/Profile'
import Notifications from '../components/Notifications/Notifications'
import Jobs from '../components/Jobs/Jobs'
import Network from '../components/Network/Network'
import Login from '../components/Login/Login'
import MainD from '../components/MainD/MainD'
import ProfilePage from '../components/ProfilePage/ProfilePage'
import Page404 from '../components/Main/Page404/Page404'
import SavedPosts from '../components/Main/SavedPosts/SavedPosts'

const MainRoutes = () => {
  return (
    <Switch>
      <Route exact path="/" render={() => <Login/>}/>
      {/* <Route exact path="/home" render={() => <Feed/>}/> */}
      <Route exact path="/home" render={() => <MainD/>}/>
      <Route exact path="/network" render={() => <Network/>}/>
      <Route exact path="/jobs" render={() => <Jobs/>}/>
      <Route exact path="/messages" render={() => <Messages/>}/>
      <Route exact path="/notifications" render={() => <Notifications/>}/>
      <Route exact path="/personal" render={() => <Profile/>}/>
      <Route exact path="/profile" render={() => <ProfilePage/>}/>
      <Route exact path="/bookmarked" render={() => <SavedPosts/>}/>
      <Route exact path="*" render={() => <Page404/>}/>
    </Switch>
  )
}

export default MainRoutes