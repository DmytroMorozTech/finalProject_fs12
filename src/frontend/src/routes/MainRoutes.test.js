import { shallow } from 'enzyme'
import React from 'react'
import MainRoutes from './MainRoutes'

const setUp = () => shallow(<MainRoutes />)

describe('should render MainRoutes', () => {
  let component
  beforeEach(() => {
    component = setUp()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
