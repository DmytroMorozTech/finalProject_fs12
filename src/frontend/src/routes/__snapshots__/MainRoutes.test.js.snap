// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MainRoutes to match snapshot 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/signup"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/signup_g"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/forgot"
    render={[Function]}
  />
  <ProtectedRoutes
    exact={true}
    path="/home"
    render={[Function]}
  />
  <ProtectedRoutes
    component={[Function]}
    exact={true}
    path="/posts/:id"
  />
  <ProtectedRoutes
    exact={true}
    path="/network"
    render={[Function]}
  />
  <ProtectedRoutes
    exact={true}
    path="/network/connections"
    render={[Function]}
  />
  <ProtectedRoutes
    exact={true}
    path="/network/following"
    render={[Function]}
  />
  <ProtectedRoutes
    exact={true}
    path="/network/followers"
    render={[Function]}
  />
  <ProtectedRoutes
    exact={true}
    path="/network/invitation_manager/received"
    render={[Function]}
  />
  <ProtectedRoutes
    exact={true}
    path="/network/invitation_manager/sent"
    render={[Function]}
  />
  <ProtectedRoutes
    exact={true}
    path="/jobs"
    render={[Function]}
  />
  <ProtectedRoutes
    exact={true}
    path="/jobs/my_jobs"
    render={[Function]}
  />
  <ProtectedRoutes
    exact={true}
    path="/messages"
    render={[Function]}
  />
  <ProtectedRoutes
    exact={true}
    path="/messages/:id"
    render={[Function]}
  />
  <ProtectedRoutes
    exact={true}
    path="/chat/:id"
    render={[Function]}
  />
  <ProtectedRoutes
    exact={true}
    path="/chat/new/:id"
    render={[Function]}
  />
  <ProtectedRoutes
    exact={true}
    path="/notifications"
    render={[Function]}
  />
  <ProtectedRoutes
    component={[Function]}
    exact={true}
    path="/profiles/:id"
  />
  <ProtectedRoutes
    component={[Function]}
    exact={true}
    path="/organization"
  />
  <ProtectedRoutes
    exact={true}
    path="/bookmarked"
    render={[Function]}
  />
  <ProtectedRoutes
    exact={true}
    path="*"
    render={[Function]}
  />
</Switch>
`;
